src/bindings/scala/api/src/main/scala/com/ctc/omega_edit/SessionImpl.scala

```diff
@@ -174,30 +174,32 @@ private[omega_edit] class SessionImpl(p: Pointer, i: FFI) extends Session {
       caseInsensitive: Boolean = false,
       limit: Option[Long] = None
   ): List[Long] = {
-    val context =
-      i.omega_search_create_context(
-        p,
-        pattern,
-        0,
-        offset,
-        length.getOrElse(0),
-        caseInsensitive
-      )
-
-    try {
-      Iterator
-        .unfold(context -> 0) { case (context, numMatches) =>
-          Option.when(
-            limit.map(numMatches < _).getOrElse(true) && i
-              .omega_search_next_match(context, 1) > 0
-          )(
-            i.omega_search_context_get_offset(
-              context
-            ) -> (context, numMatches + 1)
-          )
-        }
-        .toList
-    } finally i.omega_search_destroy_context(context)
+    i.omega_search_create_context(
+      p,
+      pattern,
```

Comment:
This method expects a Scala `String`, because in the JFR FFI it requires a `const char *pattern`. Is this a sequence of bytes in C++? We could instead invoke `omega_search_create_context_bytes` and pass bytes from a Scala `ByteString`.

Reply:
```
[error] /Users/sdell/workspaces/daffodil/omega-edit/src/bindings/scala/api/src/main/scala/com/ctc/omega_edit/SessionImpl.scala:171:20: object google is not a member of package com
[error]       pattern: com.google.protobuf.ByteString,
[error]                    ^
[error] /Users/sdell/workspaces/daffodil/omega-edit/src/bindings/scala/api/src/main/scala/com/ctc/omega_edit/api/Session.scala:25:12: object google is not a member of package com
[error] import com.google.protobuf.ByteString
[error]            ^
[error] /Users/sdell/workspaces/daffodil/omega-edit/src/bindings/scala/api/src/main/scala/com/ctc/omega_edit/api/Session.scala:75:16: not found: type ByteString
[error]       pattern: ByteString,
[error]                ^
[error] /Users/sdell/workspaces/daffodil/omega-edit/src/bindings/scala/api/src/main/scala/com/ctc/omega_edit/api/Session.scala:25:28: Unused import
[error] import com.google.protobuf.ByteString
[error]                            ^
[error] four errors found
[error] (api / Compile / compileIncremental) Compilation failed
[error] Total time: 2 s, completed Aug 23, 2022, 1:44:01 PM
```

Reply:
@arosien @scholarsmate So updated search to use `Array[Byte]` instead of string but it still fails the search and replace works with binary data and the floating viewport test